feat(payment): PAYPAL-2732 PayPalAxo Shipping step

PavlenkoM opened the pull request:
## What?
Add labels for PayPal Connect addresses

## Why?
We need to show for shoppers addresses which we get from PayPal Connect.

## Testing / Proof
BC - guest
PP Connect - guest

https://github.com/bigcommerce/checkout-js/assets/9430298/1027b77f-e1df-4c18-ac96-5bc6bf3c91da

BC - registered
PP Connect - guest

https://github.com/bigcommerce/checkout-js/assets/9430298/6894e174-03ed-4c99-bac2-7a09b3ba5bd5

BC - guest
PP Connect - registered

https://github.com/bigcommerce/checkout-js/assets/9430298/8291b50a-99bb-4f65-9c27-fe50553cdc0d

BC - registered
PP Connect - registered (with address merge)

https://github.com/bigcommerce/checkout-js/assets/9430298/aa14d130-f221-4fbb-805e-c65ee0d44c36




@bigcommerce/checkout

serhii-tkachenko reviewed (approved):
Great work!

animesh1987 commented:
Hey @PavlenkoM sorry to cause issues again, can you please breakdown this PR into smaller chunks. For example you can isolate AddressSelect changes to another PR.

animesh1987 commented on packages/core/src/app/address/PayPalAxo/PoweredByPaypalConnectLabel.scss:
```diff
@@ -0,0 +1,10 @@
+        width: 4.24rem;
+        height: 0.77rem;
+    }
+}
```
EOL

bc-peng commented on packages/core/src/app/address/AddressSelect.tsx:
```diff
@@ -1,65 +1,25 @@
-import './AddressSelect.scss';
 import AddressSelectButton from './AddressSelectButton';
+import isEqualAddress from './isEqualAddress';
+import { PoweredByPaypalConnectLabel, usePayPalConnectAddress } from './PayPalAxo';
```
This PR looks good to me.
Does it make sense to create a PalPalAxo package?
cc @animesh1987

PavlenkoM replied:
This changes will be moved in Braintree package in future.

bc-peng replied:
Maybe we can create `Braintree` package now so that we don't forget to come back here.

serhii-tkachenko replied:
Hey @bc-peng. We created a task for that and I am planing to move this part to PayPalConnect/PayPalAxo package in i19 sprint.

Is it ok for you?

serhii-tkachenko replied:
We will clean up core package from paypal specific code as soon as possible

PavlenkoM commented:
> Hey @PavlenkoM sorry to cause issues again, can you please breakdown this PR into smaller chunks. For example you can isolate AddressSelect changes to another PR.
> 
> And then different shopper flows.

Most of changes in this PR is fixing of "unit" test and snapshot.
But ok, I will split it

bc-peng commented on packages/core/src/app/address/StaticAddress.scss:
```diff
@@ -1,5 +1,18 @@
 .checkout-address--static {
+    $icon-size: 1.85rem;
```
Sorry I missed this before.
Could you please move this setting to `packages/core/src/scss/settings/foundation/forms/_settings.scss` and provide more information in the varible's name?
Plus, I didn't see we use `1.85rem` before, can you please justify why this value is appropriate?

PavlenkoM replied:
This size is specific only to one icon PP Connect, and size 1.85 is regarding to design.
In case when it is a size only for one icon in specific case I don't think that we need to move it global settings.

bc-peng replied:
I'm aware that it's a setting just for address.
However, except in the setting file, we strongly oppose putting numbers into CSS code. By doing so, we can control all of these different sizes from a single location.
Perhaps we could name it `$paypal-connect-icon-size` and put it there.

PavlenkoM replied:
done

bc-peng commented on packages/core/src/app/address/StaticAddress.scss:
```diff
@@ -1,5 +1,18 @@
+
     .address-entry {
         margin: 0;
+        padding-right: $icon-size;
```
I undertand you want to make space for the icon.
However, this class also applies to the condition when there is no icon.
Could you please create a new CSS class for PP icon so that it won't impact existing UI?

PavlenkoM replied:
yes, you are right, it will be better, will do

PavlenkoM replied:
done

animesh1987 commented on packages/core/src/app/address/AddressSelect.tsx:
```diff
@@ -38,7 +39,7 @@ const AddressSelectMenu: FunctionComponent<AddressSelectProps> = ({
             <li className="dropdown-menu-item dropdown-menu-item--select" key={address.id}>
                 <a href="#" onClick={preventDefault(() => onSelectAddress(address))}>
-                    <StaticAddress address={address} />
+                    <StaticAddress address={address} showProviderIcon />
```
Does this mean we are always passing `showProviderIcon` as `true` for `<StaticAddress />` now?

PavlenkoM replied:
in current place - yes
need this flag because of we should show this icon not for all places when we reuse this code
and checking for show/hide PP icon inside [the component](https://github.com/bigcommerce/checkout-js/pull/1423/files#diff-ded4d841a24b30f3c3ba48ec964a53a61045c8156846829a1aab4bb7ee986ed9R49)

animesh1987 replied:
Can we not just make the decision to show the label if the address is from paypal instead of adding an optional prop? Since we check if the address is from Paypal or not, I think that check should be good enough 🤔

PavlenkoM replied:
the same components `StaticAddress` with the same addresses but one with icon and another without
<img width="753" alt="Screenshot 2023-08-16 at 15 58 21" src="https://github.com/bigcommerce/checkout-js/assets/9430298/99f893fe-3976-4e7e-9813-f03eda28aedc">
<img width="752" alt="Screenshot 2023-08-16 at 15 58 41" src="https://github.com/bigcommerce/checkout-js/assets/9430298/6896328e-a2ea-4a02-ae1c-c9e28d383b2c">

PavlenkoM replied:
PayPal approved to add icon to each places with this component, so will remove this flag

PavlenkoM commented:
Hi @animesh1987 
As we discussed I have updated this PR
• remove additional property from `StaticAddress` component
• remove logic with LocalStorage
• updated hook, for now it returns only value object and method was moved to utils

bc-peng commented on packages/core/src/scss/settings/foundation/forms/_settings.scss:
```diff
@@ -165,3 +165,5 @@ $select-outline:                     0;
 $select-paddingRight:                spacing("single");
 $select-focus-borderColor:           $input-focus-border-color;
 $select-focus-boxShadow:             $input-focus-boxShadow;
+
+$paypal-connect-icon-size:           remCalc(30px);
```
```suggestion

// We use it to display a PayPal Connect icon next to the static shipping address
$paypal-connect-icon-size:           remCalc(30px);
```

animesh1987 commented:
Nice work @PavlenkoM, thanks for the patience with this one.

bc-peng commented:
Kudos to you!!!!
